admin/src/pages/HomePage.tsx

```diff
@@ -4,15 +4,16 @@ import {InstalledPlugin, PluginDef, SearchParams} from "./Plugin.ts";
 import {useDebounce} from "../utils/useDebounce.ts";
 import {Trans, useTranslation} from "react-i18next";
 import {SearchField} from "../components/SearchField.tsx";
-import {Download, Trash} from "lucide-react";
+import {ArrowUpFromDot, Download, Trash} from "lucide-react";
 import {IconButton} from "../components/IconButton.tsx";
 import {determineSorting} from "../utils/sorting.ts";
 
 
 export const HomePage = () => {
     const pluginsSocket = useStore(state=>state.pluginsSocket)
     const [plugins,setPlugins] = useState<PluginDef[]>([])
-    const [installedPlugins, setInstalledPlugins] = useState<InstalledPlugin[]>([])
+  const installedPlugins = useStore(state=>state.installedPlugins)
+  const setInstalledPlugins = useStore(state=>state.setInstalledPlugins)
```

Comment:
I think you know what you do here, but can you explain why it is needed to have installedPlugins and setInstalledPlugins?

Reply:
Yeah. So I had it fixed that the plugins are updated but somehow everytime I visited the page it showed for an instant the list of installed plugins and then it disappeared. The problem is that the useEffect does not contain the installedPlugins in its dependency array. That way the useEffect always contained an empty list of installed plugins. You could fix that by adding installedPlugins into the dependency array but that would cause a circle. The installedPlugins will be updated but the installedPlugins cause another rerender. Later in the code there is this useStore().getState() call which extracts the current installedPlugins from the store. This does not cause a rerender because we simply call a function that returns a value. That way I broke the cycle and the installedPlugins are correctly displayed with their updatable state.